crates/common/src/events/block_placement.rs

```diff
@@ -0,0 +1,9 @@
+use libcraft_core::{BlockPosition, InteractBlockFace, InteractHand, Vec3f};
+
+pub struct BlockPlacementEvent {
+    pub hand: InteractHand,
+    pub location: BlockPosition,
+    pub face: InteractBlockFace,
+    pub cursor_position: Vec3f,
+    pub inside_block: bool,
```

Comment:
Is this if the player tried to place a block while he was inside a block?
Documentation.